packages/ui/src/common/username-field.tsx

```diff
@@ -4,9 +4,13 @@ import React from 'react';
 import * as Field from '../primitives/field';
 
 export function UsernameField({
+  alternativeFieldTrigger,
   label = 'Username',
   ...props
-}: { label?: React.ReactNode } & Omit<React.ComponentProps<typeof Common.Input>, 'type'>) {
+}: { alternativeFieldTrigger?: React.ReactNode; label?: React.ReactNode } & Omit<
+  React.ComponentProps<typeof Common.Input>,
```

Comment:
I'll work to export these types.

Reply:
Cheers! That being said, I don't see the harm in doing `React.ComponentProps<>` to infer anyway; I wouldn't consider this a dealbreaker